modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -166,7 +174,17 @@ void deleteOldChunks(String name, int firstChunk) {
     //visible for testing
     protected void updateTimestamp(String name, Metadata old) {
         logger.info("geoip database [" + name + "] is up to date, updated timestamp");
-        state = state.put(name, new Metadata(System.currentTimeMillis(), old.getFirstChunk(), old.getLastChunk(), old.getMd5()));
+        long timestamp = System.currentTimeMillis();
+        UpdateByQueryRequest request = new UpdateByQueryRequest(DATABASES_INDEX).setRefresh(true);
+        request.setScript(new Script("ctx._source.timestamp=" + timestamp + "L"));
```

Comment:
Could we record the timestamp elsewhere instead of adding a field to the chunk documents and updating all these documents? Maybe store it in the cluster state as mapping metadata or custom data in `IndexMetadata`? Under the hood this results in reindexing all the chunk documents, which would be nice if this could be avoided.

Reply:
Yeah, I've added additional field in `GeoIpTaskState.Metadata` to use in validity test instead of updating timestamp and it does simplify code greatly. Thanks for suggestion!

Reply:
That is great! 👍